ibis/pyspark/compiler.py

```diff
@@ -59,12 +59,13 @@ def translate(self, expr, scope, **kwargs):
         # The operation node type the typed expression wraps
         op = expr.op()
 
-        if op in scope:
-            return scope[op]
+        result = scope.get_value(op)
+        if result is not None:
```

Comment:
Does `if op in scope` still work?

Reply:
It works, but we will change that anyway when we add time context.
Here the motivation of `if op in scope` is testing whether we could use the existing result cached in scope. 
When we have time context, we could trust and use the cached data if: 
    1. `op` is in scope, and,
    2. The `timecontext` associated with `op` is a time context equal to, or larger than the current time context.
So we will need to test to make sure both of these conditions are true. `get_value` covers both of them.

Reply:
I find it to be confusing to see both `op in scope` and `scope.get_value(op, timecontext) is not None` in code, can explain when each of those should be used?

Reply:
Explained in docstring for `__contains__` about their usage